### PR TITLE
CNL: remove the io.data and io.wb modules 

### LINKED ISSUE #13724: DOC: remove datareader docs
`io.data` and `io.web` have been deprecated since 0.17 and moved to the `pandas-datareader` project. I am not necessarily saying we should already remove the code (although we could, certainly given that the tests cause failures from time to time, but we can also wait until 0.20), but my proposal is to already remove the docs (and only include a link to the pandas-datareader docs). 

Reason: some of the examples are failing anyways (due to issues with yahoo), our docs already take quite a while to build, and the exact same content is available in the pandas-datareader docs.

Any objections?

@jreback @davidastephens @sinhrks @femtotrader

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Closes #13724

### CODE REVIEW

File: pandas/io/data.py [hunk 1]
```diff
@@ -1,1247 +1,6 @@
...
-            frame['Underlying_Price'] = np.nan
-            frame["Quote_Time"] = np.nan
-        frame.rename(columns={'Open Int': 'Open_Int'}, inplace=True)
-        frame['Type'] = type
-        frame.set_index(['Strike', 'Expiry', 'Type', 'Symbol'], inplace=True)
-
-        return frame
+raise ImportError(
+    "The pandas.io.data module is moved to a separate package "
+    "(pandas-datareader). After installing the pandas-datareader package "
```
Comment (jreback) [inline pandas/io/data.py]: need an issue to remove this in the future :>

File: pandas/io/wb.py [hunk 2]
```diff
@@ -1,314 +1,6 @@
...
-    """
-    # Create cached list of series if it does not exist
-    global _cached_series
-    if type(_cached_series) is not pandas.core.frame.DataFrame:
-        _cached_series = get_indicators()
-    data = _cached_series[field]
-    idx = data.str.contains(string, case=case)
-    out = _cached_series.ix[idx].dropna()
-    return out
+raise ImportError(
```
Comment (jreback) [inline pandas/io/wb.py]: I would just blow this one away; no exception

File: pandas/io/data.py (hunk 1 above)
Comment (jorisvandenbossche) [inline pandas/io/data.py]: yep, added an entry in the overview at https://github.com/pydata/pandas/issues/6581

File: pandas/io/wb.py (hunk 2 above)
Comment (jorisvandenbossche) [inline pandas/io/wb.py]: It doesn't is much effort to keep it .. ? (and it's for the same reason as the import error in data.py), but indeed probably used a lot less

**Comment by jreback** [comment]
@jorisvandenbossche ok lgtm (and fine to have error on both).

**Comment by jreback** [comment]
whoo hoo!

**Comment by gfyoung** [comment]
@jorisvandenbossche : any reason why we just didn't add `pandas-datareader` as a dependency and perhaps do a `try-except` block with `import pandas-datareader`?  I ask because in the <a href="https://github.com/pydata/pandas-datareader/blob/master/README.rst">README</a> for `pandas-datareader`, it seems to imply that `pandas` should be dependent on `pandas-datareader` by now.

**Comment by jreback** [comment]
the dependency is the other way
u can submit a PR to pandas-datareader if you want to change the wording

**Comment by gfyoung** [comment]
@jreback : fair enough, but I still don't get why we didn't do `try-except` with `pandas-datareader`, or rather, we even have these files at all.

**Comment by jreback** [comment]
they raise a more helpful message

**Comment by gfyoung** [comment]
@jreback : Ah, so we're helping people with transitioning from `io.data` to `pandas-datareader`?

**Comment by jreback** [comment]
of course -

**Comment by jorisvandenbossche** [comment]
Hmm, didn't see that in the readme of pandas-datareader (and it's not how it is phrased in the pandas docs). Anyway, I see that your PR over there has already been merged. 
As I said in the issue https://github.com/pydata/pandas/issues/13724, there was some unresolved discussion about this at the time we deprecated it and created pandas-datareader (I think @hayd was in favor of doing it like you say).  

Keeping the file with the custom import error is indeed to have a more helpful error for transitioning to pandas-datareader.

**Pull request merged**